apps/gauzy/src/app/pages/goals/goal-template-select/goal-template-select.component.ts

```
@@ -150,59 +149,71 @@
-					await this.keyResultService.createBulkKeyResult(keyResults).then((res) => {
-						if (res) {
-							this.toastrService.success(this.getTranslation('TOASTR.MESSAGE.KEY_RESULTS_CREATED'));
-							this.closeDialog('done');
-						}
-					});
+				// Wait for all KPI promises to complete
+				const updatedKeyResults = await Promise.all(kpiPromises);
+
+				// Prepare Key Results for bulk creation
+				const keyResults = updatedKeyResults.map((keyResult) => {
+					delete keyResult.kpi;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)